gramps/plugins/tool/reorderids.py

```diff
@@ -548,6 +551,9 @@ def _execute(self):
                         self.progress.set_pass(
                             _('Reorder %s IDs ...') % _(prim_objs.title()),
                             self.obj_values[prim_obj].quant_id)
+                    # reset the db next_id index to zero so we restart new IDs
+                    # at lowest possible position
+                    setattr(self.db, DB_INDXES[prim_obj] + 'map_index', 0)
```

Comment:
I'm not sure that this is desirable in all cases.  The original idea of this tool was confusingly to change the format of the IDs and not actually the order.  Some users prefer not to reuse the ID of an object that has been deleted.

Reply:
@Nick-Hall I'm not sure how to deal with this; I fully understand the bug submitter who wants his IDs to be nice and contiguous after a full resort, and remain contiguous with new items.  BSDDB kind of acts that way now, in that after a restart, the new ids will fill in any gaps.  Where now, with SQLite the only way to avoid gaps is to export/import XML with new tree.

Do you think we need an option for this?

Reply:
No.  We should avoid extra options unless they are necessary

The tool doesn't guarantee an order, so we are free to change the SQLite order to conform the the BSDDB order.